app/src/main/java/com/example/swugger/EditEventDialogFragment.java

```diff
@@ -0,0 +1,313 @@
+
+    public EditEventDialogFragment() { }
+
+    @Override
+    public void onDateSet(DatePicker view, int year, int month, int dayOfMonth) {
+        mDatePicker = view;
+        // regardless of whether or not the user chose a different date, update the text
+        mDateText.setText(String.format(Locale.US, "%d/%d/%d", month+1, dayOfMonth, year));     // months+1 because months is zero-indexed
+        // change the text color to indicate whether or not there's a change
+        Resources res = getResources();
+        if (mEvent.getMonth() != month || mEvent.getDay() != dayOfMonth || mEvent.getYear() != year) {
+            mDateIcon.setColorFilter(res.getColor(R.color.colorPrimary));
+            mDateText.setTextColor(res.getColor(R.color.colorPrimary));
+        } else {
+            mDateIcon.setColorFilter(res.getColor(R.color.black));
+            mDateText.setTextColor(res.getColor(R.color.gray));
+        }
+    }
+
+    @Override
+    public void onTimeSet(TimePicker view, int hourOfDay, int minute) {
+        mTimePicker = view;
+        // regardless of whether or not the user chose a different time, update the text
+        mTimeText.setText(Event.convertHourAndMinuteToTimeStamp(hourOfDay, minute));
+        // change the text color to indicate whether or not there's a change
+        Resources res = getResources();
+        if (mEvent.getHour() != hourOfDay|| mEvent.getMinute() != minute) {
+            mTimeIcon.setColorFilter(res.getColor(R.color.colorPrimary));
+            mTimeText.setTextColor(res.getColor(R.color.colorPrimary));
+        } else {
+            mTimeIcon.setColorFilter(getResources().getColor(R.color.black));
+            mTimeText.setTextColor(getResources().getColor(R.color.gray));
+        }
+    }
+
+    /** This method should not be called when Event.MAX_NUM_REMINDERS is reached because the button should be invisible.*/
+    @Override
+    public void onPositiveClickAddReminder(Integer daysBefore, Integer hoursBefore, Integer minutesBefore) {
+        long daysMilliseconds = daysBefore.longValue() * millisecondsPerDay;      // if days is super huge (not likely), watch for overflows
+        long hoursMilliseconds = hoursBefore.longValue() * millisecondsPerHour;
+        long minutesMilliseconds = minutesBefore.longValue() * millisecondsPerMinute;
+        long totalMilliseconds = daysMilliseconds + hoursMilliseconds + minutesMilliseconds;
+
+        // Get a Calendar representation of the date given by the original event
+        Calendar eventDate = Calendar.getInstance();
+        eventDate.set(mEvent.getYear(), mEvent.getMonth(), mEvent.getDay(), mEvent.getHour(), mEvent.getMinute());
+        // Calculate the date of the reminder
+        Calendar reminderDate = Calendar.getInstance();
+        long reminderDateInMilliseconds = eventDate.getTimeInMillis() - totalMilliseconds;
+        // eventDate should ALWAYS be >= reminderDate
+        if (reminderDateInMilliseconds < 0) {
```

Comment:
For a better experience, it might be better to tell the user this rather than throw an error

Reply:
for sure, i have this here for myself right now as i develop so i can more readily see some errors

Reply:
maybe there's a better system that involves separating these things as well as testing, where i can call a separate class or something to handle it for me and then set the behavior there so i can edit more easily